helper/helper_test.go

```diff
@@ -70,3 +72,39 @@ func TestCreateECToken(t *testing.T) {
 
 	defer os.RemoveAll("dummy-folder")
 }
+
+func TestFormatUploadFilePath(t *testing.T) {
+
+	// This is non-trivial only on windows
+	unixPath := "a/b/c.c4gh"
+	testPath := filepath.Join("a", "b", "c.c4gh")
+	uploadPath, err := FormatUploadFilePath(testPath)
+	assert.NoError(t, err)
+	assert.Equal(t, unixPath, uploadPath)
+
+	testPath = "a\\b\\c.c4gh"
+	uploadPath, err = FormatUploadFilePath(testPath)
+	if runtime.GOOS == "windows" {
+		// this is expected to work on windows
+		assert.Equal(t, unixPath, uploadPath)
+		assert.NoError(t, err)
+	} else {
+		// this is expected to fail on linux/mac
+		assert.ErrorContains(t, err, "filepath contains disallowed characters: \\, \\")
+
+	}
```

Comment:
```suggestion
	testPath = "a\\b\\c.c4gh"
	uploadPath, err = FormatUploadFilePath(testPath)
	assert.ErrorContains(t, err, "filepath contains disallowed characters: \\, \\")
```

That test is only viable if the test suite is executed on a windows machine.

Reply:
Yes, indeed. However I added it there because,

1. I expect that someone may wish to run the testsuite from a windows machine
2. In a way, this does provide a way to test a scenario where a windows client interacts with the `proxy` since the different   behavior documented here originates from the way `testPath` is interpreted depending on the OS thus simulating the s3 client's host OS.
3. I think keeping these tests is a good way to document these differences between win and linux